### PR TITLE
ci(ai-client): continuous dev deployment on edge merge

### PR DESCRIPTION
**Pull request by y3rsh**
# Overview

When anything in `opentrons-ai-client` changes and is push (merge) to `edge`, build and deploy.

## Notes

- did not parameterize as we are not sure yet if we even will like this or how we will deploy to higher environments
- the future holds end to end tests but we may use DataDog monitors and synthetic test 

## Tasks

- [x] secrets added to the repository

### CODE REVIEW

**Review by koji — APPROVED** [review]
LGTM!

File: .github/workflows/opentrons-ai-client-dev-continuos-deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+          key: js-${{ secrets.GH_CACHE_VERSION }}-${{ runner.os }}-yarn-${{ hashFiles('yarn.lock') }}
+      - name: 'setup-js'
+        run: |
+          npm config set cache ${{ github.workspace }}/.npm-cache
+          yarn config set cache-folder ${{ github.workspace }}/.yarn-cache
+          make setup-js
+      - name: build
+        run: |
+          make -C opentrons-ai-client build
+      - name: Configure AWS Credentials
```
Comment (Elyorcv) [inline .github/workflows/opentrons-ai-client-dev-continuos-deploy.yaml:57]: new here. I notice that most names are a single quotes `'`. Just for the consistency. 
```suggestion
      - name: 'Configure AWS Credentials'
```

File: .github/workflows/opentrons-ai-client-dev-continuos-deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+          path: |
+            ${{ github.workspace }}/.npm-cache/_prebuild
+            ${{ github.workspace }}/.yarn-cache
+          key: js-${{ secrets.GH_CACHE_VERSION }}-${{ runner.os }}-yarn-${{ hashFiles('yarn.lock') }}
+      - name: 'setup-js'
+        run: |
+          npm config set cache ${{ github.workspace }}/.npm-cache
+          yarn config set cache-folder ${{ github.workspace }}/.yarn-cache
+          make setup-js
+      - name: build
```
Comment (Elyorcv) [inline .github/workflows/opentrons-ai-client-dev-continuos-deploy.yaml]: new here. I notice that most names are a single quotes '. Just for the consistency.
```suggestion
      - name: 'build'
```

**Review by Elyorcv — COMMENTED** [review]
LGTM

**Review by shlokamin — APPROVED** [review]
🥳 thank you @y3rsh!

**Pull request merged**